salt/modules/boto_ec2.py

```diff
@@ -1914,6 +1914,8 @@ def detach_volume(volume_id, instance_id=None, device=None, force=False,
                  only as a last resort to detach a volume from a failed instance. The instance
                  will not have an opportunity to flush file system caches nor file system meta data.
                  If you use this option, you must perform file system check and repair procedures.
+    wait_to_finish
+       (int) - The time to wait for detaching volume complete, in second.
```

Comment:
From a consistency perspective, the other boto modules use wait_for_sync, which is a bool that waits until it's finished, not based on a timeout.